### PR TITLE
Add support for mysqldump

### PR DESCRIPTION
**Pull request by shenli**
Support some mysql syntax to prevent mysqldump error.
We support mysqldump now.
Fix https://github.com/pingcap/tidb/issues/520

### CODE REVIEW

File: mysql/time.go [hunk 1]
```diff
@@ -107,6 +107,11 @@ type Time struct {
 	Fsp int
 }
 
+// GetCurrentTime returns current time with type tp.
+func GetCurrentTime(tp uint8) Time {
```
Comment (ngaut) [inline mysql/time.go]: s/GetCurrentTime/CurrentTime

File: plan/plans/show.go [hunk 2]
```diff
@@ -111,6 +120,11 @@ func (s *ShowPlan) GetFields() []*field.ResultField {
 		names = []string{"Table", "Create Table"}
 	case stmt.ShowGrants:
 		names = []string{fmt.Sprintf("Grants for %s", s.User)}
+	case stmt.ShowTriggers:
+		names = []string{"Trigger", "Event", "Table", "Statement", "Timing", "Created",
+			"sql_mode", "Definer", "character_set_client", "collation_connection", "Database Collation"}
```
Comment (siddontang) [inline plan/plans/show.go]: I guess Sql_mode?

File: parser/parser_test.go [hunk 3]
```diff
@@ -39,7 +39,7 @@ func (s *testParserSuite) TestSimple(c *C) {
 		"start", "global", "tables", "text", "time", "timestamp", "transaction", "truncate", "unknown",
 		"value", "warnings", "year", "now", "substring", "mode", "any", "some", "user", "identified",
 		"collation", "comment", "avg_row_length", "checksum", "compression", "connection", "key_block_size",
-		"max_rows", "min_rows", "national", "row", "quarter", "escape", "grants",
+		"max_rows", "min_rows", "national", "row", "quarter", "escape", "grants", "status", "FIELDS", "TRIGGERS",
```
Comment (qiuyesuifeng) [inline parser/parser_test.go]: lower case？

File: plan/plans/show.go (hunk 2 above)
Comment (shenli) [inline plan/plans/show.go]: https://dev.mysql.com/doc/refman/5.7/en/show-triggers.html
I copy the column names from here. It is "sql_mode".

File: parser/parser_test.go (hunk 3 above)
Comment (shenli) [inline parser/parser_test.go]: PTAL

File: plan/plans/show.go (hunk 2 above)
Comment (siddontang) [inline plan/plans/show.go]: got it, seems no unify here.

**Comment by qiuyesuifeng** [comment]
LGTM

**Comment by siddontang** [comment]
LGTM

**Pull request merged**